src/main/java/Duke.java

```diff
@@ -1,10 +1,377 @@
+        try {
+            fr = new FileWriter(file);
+            fr.write(lines.toString());
+        } catch (IOException e) {
+            e.printStackTrace();
+        } finally {
+            // close resources
+            try {
+                if (fr != null) {
+                    fr.close();
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+                System.out.println(MESSAGE_ERROR_FILE);
+            }
+        }
+    }
+
+    public static void delete(String inputCommand) {
+        int taskNumber;
+        Task selectedTask;
+
+        try{
+            taskNumber = Integer.parseInt(removeFirstWordOf(inputCommand)) - 1;
+        } catch (NumberFormatException e) {
+            System.out.println(MESSAGE_ERROR_DELETE);
+            return;
+        }
+
+        try{
+            selectedTask = taskList.get(taskNumber);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("ERROR: task '" + (taskNumber + 1) + "' does not exist");
+            return;
+        }
+
+        System.out.println("Noted. I've removed this task:\n\t" + selectedTask.toString());
+        System.out.println("Now you have " + (--numberOfTasks) + " tasks in the list.");
+        taskList.remove(taskNumber);
+    }
+
+    public static Task parseCommand(String inputCommand) {
+        Task task = null;
+        String firstWord = getFirstWordOf(inputCommand);
+
+        try{
+            switch(firstWord){
+            case "list":
+                try{
+                    printList();
+                } catch(DukeException e){
+                    System.out.println(MESSAGE_ERROR_EMPTY_LIST);
+                }
+                break;
+            case "done":
+                markDone(inputCommand);
+                break;
+            case "todo":
+                try{
+                    task = addTodo(inputCommand);
+                } catch(DukeException e) {
+                    System.out.println(MESSAGE_ERROR_TODO);
+                }
+                break;
+            case "deadline":
+                try{
+                    task = addDeadline(inputCommand);
+                } catch(DukeException e) {
+                    System.out.println(MESSAGE_ERROR_DEADLINE);
+                }
+                break;
+            case "event":
+                try{
+                    task = addEvent(inputCommand);
+                } catch(DukeException e) {
+                    System.out.println(MESSAGE_ERROR_EVENT);
+                }
+                break;
+            case "delete":
+                delete(inputCommand);
+                break;
+            case "bye":
+                return null;
+            default:
+                throw new DukeException();
+            }
+            save();
+        } catch (DukeException e){
+            System.out.println("ERROR: '" + firstWord + "'" + MESSAGE_ERROR_INVALID_COMMAND);
+        }
+        return task;
+    }
+
```

Comment:
Just suggestion: the Duke.java file is probably too long. You may consider reducing it by a class for managing files and a class for managing tasks.